### ISSUE TITLE
install problems on Ubuntu 22.04.4 LTS (Jammy Jellyfish)

### ISSUE THREAD

**Issue by katestange** [numberscope/backscope#137]
I'm having trouble installing backscope (install-requirements.sh) after upgrading to Ubuntu 22.04.4.  I'm pasting the error (with cypari, big surprise).  I'm using python 3.10.12.  Any ideas?  I'm posting as an issue since if we solve this we'll perhaps want to update the docs.

```
Building wheels for collected packages: cypari2, jsonify
  Building wheel for cypari2 (pyproject.toml) ... error
  error: subprocess-exited-with-error
  
  × Building wheel for cypari2 (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [261 lines of output]
      running bdist_wheel
      running build
      running build_py
      creating build
      creating build/lib.linux-x86_64-cpython-310
      creating build/lib.linux-x86_64-cpython-310/cypari2
      copying cypari2/__init__.py -> build/lib.linux-x86_64-cpython-310/cypari2
      copying cypari2/stack.pxd -> build/lib.linux-x86_64-cpython-310/cypari2
      copying cypari2/closure.pxd -> build/lib.linux-x86_64-cpython-310/cypari2
      copying cypari2/paripriv.pxd -> build/lib.linux-x86_64-cpython-310/cypari2
      copying cypari2/gen.pxd -> build/lib.linux-x86_64-cpython-310/cypari2
      copying cypari2/string_utils.pxd -> build/lib.linux-x86_64-cpython-310/cypari2
      copying cypari2/convert.pxd -> build/lib.linux-x86_64-cpython-310/cypari2
      copying cypari2/paridecl.pxd -> build/lib.linux-x86_64-cpython-310/cypari2
      copying cypari2/handle_error.pxd -> build/lib.linux-x86_64-cpython-310/cypari2
      copying cypari2/auto_paridecl.pxd -> build/lib.linux-x86_64-cpython-310/cypari2
      copying cypari2/pari_instance.pxd -> build/lib.linux-x86_64-cpython-310/cypari2
      copying cypari2/types.pxd -> build/lib.linux-x86_64-cpython-310/cypari2
      copying cypari2/Py_SET_SIZE.h -> build/lib.linux-x86_64-cpython-310/cypari2
      copying cypari2/cypari.h -> build/lib.linux-x86_64-cpython-310/cypari2
      running build_ext
      
      Error compiling Cython file:
      ------------------------------------------------------------
      ...
      #                  http://www.gnu.org/licenses/
      #*****************************************************************************
      
      from __future__ import absolute_import, division, print_function
      
      from cysignals.signals cimport sig_on, sig_off, sig_block, sig_unblock, sig_error
      ^
      ------------------------------------------------------------
      
      cypari2/closure.pyx:36:0: 'cysignals/signals.pxd' not found
      
      Error compiling Cython file:
      ------------------------------------------------------------
      ...
      #                  http://www.gnu.org/licenses/
      #*****************************************************************************
      
      from __future__ import absolute_import, division, print_function
      
      from cysignals.signals cimport sig_on, sig_off, sig_block, sig_unblock, sig_error
      ^
      ------------------------------------------------------------
      
      cypari2/closure.pyx:36:0: 'cysignals/signals/sig_on.pxd' not found
      
      Error compiling Cython file:
      ------------------------------------------------------------
      ...
      #                  http://www.gnu.org/licenses/
      #*****************************************************************************
      
      from __future__ import absolute_import, division, print_function
      
      from cysignals.signals cimport sig_on, sig_off, sig_block, sig_unblock, sig_error
      ^
      ------------------------------------------------------------
      
      cypari2/closure.pyx:36:0: 'cysignals/signals/sig_off.pxd' not found
      
      Error compiling Cython file:
      ------------------------------------------------------------
      ...
      #                  http://www.gnu.org/licenses/
      #*****************************************************************************
      
      from __future__ import absolute_import, division, print_function
      
      from cysignals.signals cimport sig_on, sig_off, sig_block, sig_unblock, sig_error
      ^
      ------------------------------------------------------------
      
      cypari2/closure.pyx:36:0: 'cysignals/signals/sig_block.pxd' not found
      
      Error compiling Cython file:
      ------------------------------------------------------------
      ...
      #                  http://www.gnu.org/licenses/
      #*****************************************************************************
      
      from __future__ import absolute_import, division, print_function
      
      from cysignals.signals cimport sig_on, sig_off, sig_block, sig_unblock, sig_error
      ^
      ------------------------------------------------------------
      
      cypari2/closure.pyx:36:0: 'cysignals/signals/sig_unblock.pxd' not found
      
      Error compiling Cython file:
      ------------------------------------------------------------
      ...
      #                  http://www.gnu.org/licenses/
      #*****************************************************************************
      
      from __future__ import absolute_import, division, print_function
      
      from cysignals.signals cimport sig_on, sig_off, sig_block, sig_unblock, sig_error
      ^
      ------------------------------------------------------------
      
      cypari2/closure.pyx:36:0: 'cysignals/signals/sig_error.pxd' not found
      
      Error compiling Cython file:
      ------------------------------------------------------------
      ...
          It has 5 optional ``GEN``s as argument, a ``nargs`` argument
          specifying how many arguments are valid and one ``ulong``, which is
          actually a Python callable object cast to ``ulong``.
          """
          if nargs > 5:
              sig_error()
              ^
      ------------------------------------------------------------
      
      cypari2/closure.pyx:112:8: 'sig_error' is not a constant, variable or function identifier
      
      Error compiling Cython file:
      ------------------------------------------------------------
      ...
          args[2] = arg3
          args[3] = arg4
          args[4] = arg5
          args[nargs] = NULL
      
          sig_block()
          ^
      ------------------------------------------------------------
      
      cypari2/closure.pyx:123:4: 'sig_block' is not a constant, variable or function identifier
      
      Error compiling Cython file:
      ------------------------------------------------------------
      ...
          # Disallow interrupts during the Python code inside
          # call_python_func_impl(). We need to do this because this function
          # is very likely called within sig_on() and interrupting arbitrary
          # Python code is bad.
          cdef GEN r = call_python_func(args, <PyObject*>py_func)
          sig_unblock()
          ^
      ------------------------------------------------------------
      
      cypari2/closure.pyx:129:4: 'sig_unblock' is not a constant, variable or function identifier
      
      Error compiling Cython file:
      ------------------------------------------------------------
      ...
          # is very likely called within sig_on() and interrupting arbitrary
          # Python code is bad.
          cdef GEN r = call_python_func(args, <PyObject*>py_func)
          sig_unblock()
          if not r:  # An exception was raised
              sig_error()
              ^
      ------------------------------------------------------------
      
      cypari2/closure.pyx:131:8: 'sig_error' is not a constant, variable or function identifier
      
      Error compiling Cython file:
      ------------------------------------------------------------
      ...
      
      # Install the function "call_python" for use in the PARI library.
      cdef entree* ep_call_python
      
      cdef int _pari_init_closure() except -1:
          sig_on()
          ^
      ------------------------------------------------------------
      
      cypari2/closure.pyx:139:4: 'sig_on' is not a constant, variable or function identifier
      
      Error compiling Cython file:
      ------------------------------------------------------------
      ...
      
      cdef int _pari_init_closure() except -1:
          sig_on()
          global ep_call_python
          ep_call_python = install(<void*>call_python, "call_python", 'DGDGDGDGDGD5,U,U')
          sig_off()
          ^
      ------------------------------------------------------------
      
      cypari2/closure.pyx:142:4: 'sig_off' is not a constant, variable or function identifier
      
      Error compiling Cython file:
      ------------------------------------------------------------
      ...
          # Only 5 arguments are supported for now...
          if nargs > 5:
              nargs = 5
      
          # Fill in default arguments of PARI function
          sig_on()
          ^
      ------------------------------------------------------------
      
      cypari2/closure.pyx:230:4: 'sig_on' is not a constant, variable or function identifier
      Compiling cypari2/closure.pyx because it changed.
      Compiling cypari2/convert.pyx because it changed.
      Compiling cypari2/gen.pyx because it changed.
      Compiling cypari2/handle_error.pyx because it changed.
      Compiling cypari2/pari_instance.pyx because it changed.
      Compiling cypari2/stack.pyx because it changed.
      Compiling cypari2/string_utils.pyx because it changed.
      [1/7] Cythonizing cypari2/closure.pyx
      Traceback (most recent call last):
        File "/home/katestange/data/projects/numberscope/frontscope-backscope/backscope/.venv/lib/python3.10/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
          main()
        File "/home/katestange/data/projects/numberscope/frontscope-backscope/backscope/.venv/lib/python3.10/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
        File "/home/katestange/data/projects/numberscope/frontscope-backscope/backscope/.venv/lib/python3.10/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 251, in build_wheel
          return _build_backend().build_wheel(wheel_directory, config_settings,
        File "/tmp/pip-build-env-4rav3jn4/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 415, in build_wheel
          return self._build_with_temp_dir(
        File "/tmp/pip-build-env-4rav3jn4/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 397, in _build_with_temp_dir
          self.run_setup()
        File "/tmp/pip-build-env-4rav3jn4/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 497, in run_setup
          super().run_setup(setup_script=setup_script)
        File "/tmp/pip-build-env-4rav3jn4/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 313, in run_setup
          exec(code, locals())
        File "<string>", line 66, in <module>
        File "/tmp/pip-build-env-4rav3jn4/overlay/lib/python3.10/site-packages/setuptools/__init__.py", line 103, in setup
          return distutils.core.setup(**attrs)
        File "/tmp/pip-build-env-4rav3jn4/overlay/lib/python3.10/site-packages/setuptools/_distutils/core.py", line 184, in setup
          return run_commands(dist)
        File "/tmp/pip-build-env-4rav3jn4/overlay/lib/python3.10/site-packages/setuptools/_distutils/core.py", line 200, in run_commands
          dist.run_commands()
        File "/tmp/pip-build-env-4rav3jn4/overlay/lib/python3.10/site-packages/setuptools/_distutils/dist.py", line 969, in run_commands
          self.run_command(cmd)
        File "/tmp/pip-build-env-4rav3jn4/overlay/lib/python3.10/site-packages/setuptools/dist.py", line 976, in run_command
          super().run_command(command)
        File "/tmp/pip-build-env-4rav3jn4/overlay/lib/python3.10/site-packages/setuptools/_distutils/dist.py", line 988, in run_command
          cmd_obj.run()
        File "/tmp/pip-build-env-4rav3jn4/overlay/lib/python3.10/site-packages/setuptools/command/bdist_wheel.py", line 373, in run
          self.run_command("build")
        File "/tmp/pip-build-env-4rav3jn4/overlay/lib/python3.10/site-packages/setuptools/_distutils/cmd.py", line 316, in run_command
          self.distribution.run_command(command)
        File "/tmp/pip-build-env-4rav3jn4/overlay/lib/python3.10/site-packages/setuptools/dist.py", line 976, in run_command
          super().run_command(command)
        File "/tmp/pip-build-env-4rav3jn4/overlay/lib/python3.10/site-packages/setuptools/_distutils/dist.py", line 988, in run_command
          cmd_obj.run()
        File "/tmp/pip-build-env-4rav3jn4/overlay/lib/python3.10/site-packages/setuptools/_distutils/command/build.py", line 132, in run
          self.run_command(cmd_name)
        File "/tmp/pip-build-env-4rav3jn4/overlay/lib/python3.10/site-packages/setuptools/_distutils/cmd.py", line 316, in run_command
          self.distribution.run_command(command)
        File "/tmp/pip-build-env-4rav3jn4/overlay/lib/python3.10/site-packages/setuptools/dist.py", line 976, in run_command
          super().run_command(command)
        File "/tmp/pip-build-env-4rav3jn4/overlay/lib/python3.10/site-packages/setuptools/_distutils/dist.py", line 988, in run_command
          cmd_obj.run()
        File "<string>", line 46, in run
        File "/tmp/pip-build-env-4rav3jn4/normal/lib/python3.10/site-packages/Cython/Build/Dependencies.py", line 1154, in cythonize
          cythonize_one(*args)
        File "/tmp/pip-build-env-4rav3jn4/normal/lib/python3.10/site-packages/Cython/Build/Dependencies.py", line 1321, in cythonize_one
          raise CompileError(None, pyx_file)
      Cython.Compiler.Errors.CompileError: cypari2/closure.pyx
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for cypari2
  Building wheel for jsonify (pyproject.toml) ... done
  Created wheel for jsonify: filename=jsonify-0.5-py3-none-any.whl size=1538 sha256=43d011d52c70fb108070ead7b2e38a4423eed66523082ddae06ded16988c17b3
  Stored in directory: /home/katestange/.cache/pip/wheels/4e/6d/b2/9cb590fb5654ff13457f49d4aa5b16469419681613c46a2ec1
Successfully built jsonify
Failed to build cypari2
ERROR: ERROR: Failed to build installable wheels for some pyproject.toml based projects (cypari2)
Failed to install remaining requirements
```

**Comment by gwhitney** [comment]
OK, just to check, you followed https://github.com/numberscope/backscope/blob/main/doc/install-ubuntu.md including installing libpari-dev, correct? I will post again soon with some other items.

**Comment by katestange** [comment]
Correct, I have version 2.13.3-1.

**Comment by gwhitney** [comment]
OK, I recommend that you activate your virtual environment, and then try
`pip install --force cysignals==1.11.4` (I expect that will succeed) and then try `pip install --force cypari2==2.1.5`. If the latter fails, then try `pip install cypari2==2.5.4`. If either of those two succeeds, then try to go back to the step in the install-ubuntu directions that failed (presumably it was `sh tools/install-requirements.sh` and re-execute that step. If it now works, see if you can run everything OK.

Either way let us know your status back here and if it's still failing, post the transcripts of the two failed attempts to install cypari2, or whatever else it is that's failing.

**Comment by katestange** [comment]
Ok, `pip install --force cypari2==2.1.5` succeeds, so I did `sh tools/install-requirements.sh` the tail of which is this failure:
```
Building wheels for collected packages: cypari2
  Building wheel for cypari2 (pyproject.toml) ... error
  error: subprocess-exited-with-error
  
  × Building wheel for cypari2 (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [261 lines of output]
      running bdist_wheel
      running build
      running build_py
      creating build
      creating build/lib.linux-x86_64-cpython-310
      creating build/lib.linux-x86_64-cpython-310/cypari2
      copying cypari2/__init__.py -> build/lib.linux-x86_64-cpython-310/cypari2
      copying cypari2/stack.pxd -> build/lib.linux-x86_64-cpython-310/cypari2
      copying cypari2/closure.pxd -> build/lib.linux-x86_64-cpython-310/cypari2
      copying cypari2/paripriv.pxd -> build/lib.linux-x86_64-cpython-310/cypari2
      copying cypari2/gen.pxd -> build/lib.linux-x86_64-cpython-310/cypari2
      copying cypari2/string_utils.pxd -> build/lib.linux-x86_64-cpython-310/cypari2
      copying cypari2/convert.pxd -> build/lib.linux-x86_64-cpython-310/cypari2
      copying cypari2/paridecl.pxd -> build/lib.linux-x86_64-cpython-310/cypari2
      copying cypari2/handle_error.pxd -> build/lib.linux-x86_64-cpython-310/cypari2
      copying cypari2/auto_paridecl.pxd -> build/lib.linux-x86_64-cpython-310/cypari2
      copying cypari2/pari_instance.pxd -> build/lib.linux-x86_64-cpython-310/cypari2
      copying cypari2/types.pxd -> build/lib.linux-x86_64-cpython-310/cypari2
      copying cypari2/Py_SET_SIZE.h -> build/lib.linux-x86_64-cpython-310/cypari2
      copying cypari2/cypari.h -> build/lib.linux-x86_64-cpython-310/cypari2
      running build_ext
      
      Error compiling Cython file:
      ------------------------------------------------------------
      ...
      #                  http://www.gnu.org/licenses/
      #*****************************************************************************
      
      from __future__ import absolute_import, division, print_function
      
      from cysignals.signals cimport sig_on, sig_off, sig_block, sig_unblock, sig_error
      ^
      ------------------------------------------------------------
      
      cypari2/closure.pyx:36:0: 'cysignals/signals.pxd' not found
      
      Error compiling Cython file:
      ------------------------------------------------------------
      ...
      #                  http://www.gnu.org/licenses/
      #*****************************************************************************
      
      from __future__ import absolute_import, division, print_function
      
      from cysignals.signals cimport sig_on, sig_off, sig_block, sig_unblock, sig_error
      ^
      ------------------------------------------------------------
      
      cypari2/closure.pyx:36:0: 'cysignals/signals/sig_on.pxd' not found
      
      Error compiling Cython file:
      ------------------------------------------------------------
      ...
      #                  http://www.gnu.org/licenses/
      #*****************************************************************************
      
      from __future__ import absolute_import, division, print_function
      
      from cysignals.signals cimport sig_on, sig_off, sig_block, sig_unblock, sig_error
      ^
      ------------------------------------------------------------
      
      cypari2/closure.pyx:36:0: 'cysignals/signals/sig_off.pxd' not found
      
      Error compiling Cython file:
      ------------------------------------------------------------
      ...
      #                  http://www.gnu.org/licenses/
      #*****************************************************************************
      
      from __future__ import absolute_import, division, print_function
      
      from cysignals.signals cimport sig_on, sig_off, sig_block, sig_unblock, sig_error
      ^
      ------------------------------------------------------------
      
      cypari2/closure.pyx:36:0: 'cysignals/signals/sig_block.pxd' not found
      
      Error compiling Cython file:
      ------------------------------------------------------------
      ...
      #                  http://www.gnu.org/licenses/
      #*****************************************************************************
      
      from __future__ import absolute_import, division, print_function
      
      from cysignals.signals cimport sig_on, sig_off, sig_block, sig_unblock, sig_error
      ^
      ------------------------------------------------------------
      
      cypari2/closure.pyx:36:0: 'cysignals/signals/sig_unblock.pxd' not found
      
      Error compiling Cython file:
      ------------------------------------------------------------
      ...
      #                  http://www.gnu.org/licenses/
      #*****************************************************************************
      
      from __future__ import absolute_import, division, print_function
      
      from cysignals.signals cimport sig_on, sig_off, sig_block, sig_unblock, sig_error
      ^
      ------------------------------------------------------------
      
      cypari2/closure.pyx:36:0: 'cysignals/signals/sig_error.pxd' not found
      
      Error compiling Cython file:
      ------------------------------------------------------------
      ...
          It has 5 optional ``GEN``s as argument, a ``nargs`` argument
          specifying how many arguments are valid and one ``ulong``, which is
          actually a Python callable object cast to ``ulong``.
          """
          if nargs > 5:
              sig_error()
              ^
      ------------------------------------------------------------
      
      cypari2/closure.pyx:112:8: 'sig_error' is not a constant, variable or function identifier
      
      Error compiling Cython file:
      ------------------------------------------------------------
      ...
          args[2] = arg3
          args[3] = arg4
          args[4] = arg5
          args[nargs] = NULL
      
          sig_block()
          ^
      ------------------------------------------------------------
      
      cypari2/closure.pyx:123:4: 'sig_block' is not a constant, variable or function identifier
      
      Error compiling Cython file:
      ------------------------------------------------------------
      ...
          # Disallow interrupts during the Python code inside
          # call_python_func_impl(). We need to do this because this function
          # is very likely called within sig_on() and interrupting arbitrary
          # Python code is bad.
          cdef GEN r = call_python_func(args, <PyObject*>py_func)
          sig_unblock()
          ^
      ------------------------------------------------------------
      
      cypari2/closure.pyx:129:4: 'sig_unblock' is not a constant, variable or function identifier
      
      Error compiling Cython file:
      ------------------------------------------------------------
      ...
          # is very likely called within sig_on() and interrupting arbitrary
          # Python code is bad.
          cdef GEN r = call_python_func(args, <PyObject*>py_func)
          sig_unblock()
          if not r:  # An exception was raised
              sig_error()
              ^
      ------------------------------------------------------------
      
      cypari2/closure.pyx:131:8: 'sig_error' is not a constant, variable or function identifier
      
      Error compiling Cython file:
      ------------------------------------------------------------
      ...
      
      # Install the function "call_python" for use in the PARI library.
      cdef entree* ep_call_python
      
      cdef int _pari_init_closure() except -1:
          sig_on()
          ^
      ------------------------------------------------------------
      
      cypari2/closure.pyx:139:4: 'sig_on' is not a constant, variable or function identifier
      
      Error compiling Cython file:
      ------------------------------------------------------------
      ...
      
      cdef int _pari_init_closure() except -1:
          sig_on()
          global ep_call_python
          ep_call_python = install(<void*>call_python, "call_python", 'DGDGDGDGDGD5,U,U')
          sig_off()
          ^
      ------------------------------------------------------------
      
      cypari2/closure.pyx:142:4: 'sig_off' is not a constant, variable or function identifier
      
      Error compiling Cython file:
      ------------------------------------------------------------
      ...
          # Only 5 arguments are supported for now...
          if nargs > 5:
              nargs = 5
      
          # Fill in default arguments of PARI function
          sig_on()
          ^
      ------------------------------------------------------------
      
      cypari2/closure.pyx:230:4: 'sig_on' is not a constant, variable or function identifier
      Compiling cypari2/closure.pyx because it changed.
      Compiling cypari2/convert.pyx because it changed.
      Compiling cypari2/gen.pyx because it changed.
      Compiling cypari2/handle_error.pyx because it changed.
      Compiling cypari2/pari_instance.pyx because it changed.
      Compiling cypari2/stack.pyx because it changed.
      Compiling cypari2/string_utils.pyx because it changed.
      [1/7] Cythonizing cypari2/closure.pyx
      Traceback (most recent call last):
        File "/home/katestange/data/projects/numberscope/frontscope-backscope/backscope/.venv/lib/python3.10/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
          main()
        File "/home/katestange/data/projects/numberscope/frontscope-backscope/backscope/.venv/lib/python3.10/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
        File "/home/katestange/data/projects/numberscope/frontscope-backscope/backscope/.venv/lib/python3.10/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 251, in build_wheel
          return _build_backend().build_wheel(wheel_directory, config_settings,
        File "/tmp/pip-build-env-kr5mqmr1/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 415, in build_wheel
          return self._build_with_temp_dir(
        File "/tmp/pip-build-env-kr5mqmr1/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 397, in _build_with_temp_dir
          self.run_setup()
        File "/tmp/pip-build-env-kr5mqmr1/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 497, in run_setup
          super().run_setup(setup_script=setup_script)
        File "/tmp/pip-build-env-kr5mqmr1/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 313, in run_setup
          exec(code, locals())
        File "<string>", line 66, in <module>
        File "/tmp/pip-build-env-kr5mqmr1/overlay/lib/python3.10/site-packages/setuptools/__init__.py", line 103, in setup
          return distutils.core.setup(**attrs)
        File "/tmp/pip-build-env-kr5mqmr1/overlay/lib/python3.10/site-packages/setuptools/_distutils/core.py", line 184, in setup
          return run_commands(dist)
        File "/tmp/pip-build-env-kr5mqmr1/overlay/lib/python3.10/site-packages/setuptools/_distutils/core.py", line 200, in run_commands
          dist.run_commands()
        File "/tmp/pip-build-env-kr5mqmr1/overlay/lib/python3.10/site-packages/setuptools/_distutils/dist.py", line 969, in run_commands
          self.run_command(cmd)
        File "/tmp/pip-build-env-kr5mqmr1/overlay/lib/python3.10/site-packages/setuptools/dist.py", line 976, in run_command
          super().run_command(command)
        File "/tmp/pip-build-env-kr5mqmr1/overlay/lib/python3.10/site-packages/setuptools/_distutils/dist.py", line 988, in run_command
          cmd_obj.run()
        File "/tmp/pip-build-env-kr5mqmr1/overlay/lib/python3.10/site-packages/setuptools/command/bdist_wheel.py", line 373, in run
          self.run_command("build")
        File "/tmp/pip-build-env-kr5mqmr1/overlay/lib/python3.10/site-packages/setuptools/_distutils/cmd.py", line 316, in run_command
          self.distribution.run_command(command)
        File "/tmp/pip-build-env-kr5mqmr1/overlay/lib/python3.10/site-packages/setuptools/dist.py", line 976, in run_command
          super().run_command(command)
        File "/tmp/pip-build-env-kr5mqmr1/overlay/lib/python3.10/site-packages/setuptools/_distutils/dist.py", line 988, in run_command
          cmd_obj.run()
        File "/tmp/pip-build-env-kr5mqmr1/overlay/lib/python3.10/site-packages/setuptools/_distutils/command/build.py", line 132, in run
          self.run_command(cmd_name)
        File "/tmp/pip-build-env-kr5mqmr1/overlay/lib/python3.10/site-packages/setuptools/_distutils/cmd.py", line 316, in run_command
          self.distribution.run_command(command)
        File "/tmp/pip-build-env-kr5mqmr1/overlay/lib/python3.10/site-packages/setuptools/dist.py", line 976, in run_command
          super().run_command(command)
        File "/tmp/pip-build-env-kr5mqmr1/overlay/lib/python3.10/site-packages/setuptools/_distutils/dist.py", line 988, in run_command
          cmd_obj.run()
        File "<string>", line 46, in run
        File "/tmp/pip-build-env-kr5mqmr1/normal/lib/python3.10/site-packages/Cython/Build/Dependencies.py", line 1154, in cythonize
          cythonize_one(*args)
        File "/tmp/pip-build-env-kr5mqmr1/normal/lib/python3.10/site-packages/Cython/Build/Dependencies.py", line 1321, in cythonize_one
          raise CompileError(None, pyx_file)
      Cython.Compiler.Errors.CompileError: cypari2/closure.pyx
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for cypari2
Failed to build cypari2
ERROR: ERROR: Failed to build installable wheels for some pyproject.toml based projects (cypari2)
```

**Comment by gwhitney** [comment]
Hmm, that's very strange behavior. I guess after activating the venv and a successful run of `pip install --force cysignals==1.11.4` and `pip install --force cypari2=2.1.5`, you should try `pip install -r requirements-freeze.txt`. I don't expect it to succeed, but if it does, you should then be able to continue with installing and running backscope  from the step _after_ `sh tools/install-requirements.sh`.

What I expect is that this third pip command will fail with the same transcript as you just posted above (I don't know why it's failing, just that it is).

If that happens, then if you just want to get going with backscope, the next thing to try is to make your own copy of the file requirements-freeze.txt.  Let's suppose you call it `requirements-kate.txt`. In your copy, delete the two lines
```
cypari2==2.1.5
cysignals==1.11.4
```
Then do the manual direct forced install of cysignals and cypari2 that you said succeeded. When those are done, do `pip install -r requirements-kate.txt`. I expect that method will succeed. This is how we used to install cypari -- in fact, this roundabout route is the whole reason tools/install-rqeuirements.sh was written in the first place. If that installation all works, then again you should be able to continue with the _next_ step in the installation instructions.

Let us know how it goes.

If the primary goal is to have foolproof installation instructions/scripts, then of course we will need to dig a bit deeper but it might need to be a live joint debugging session rather than a github discussion, for bandwidth reasons ;-)

Fingers crossed.

**Comment by katestange** [comment]
Thank you Glen!  Thanks to your tips, I figured out that something was out of date: a mismatch between cypari2 2.1.3 and 2.1.5.  The requirements script was trying to install the wrong one.  I think it's my fault, because when I pulled again, everything went smoothly, so I must have gotten behind the repository or overwritten the requirements file or something.  I'm sorry!  I don't know how that happened, because I thought I was typically keeping up to date, and the switch to 2.1.5 was back in december and I remember messing with it then.  The operating system upgrade should not have "reverted" me, but somewhere along the way I must have done something out of order or misunderstood git.  (The operating system upgrade problems did seem to relate to github's CLI?)  But in any case, the bottom line is fairly clear:  that this was my fault and shouldn't occur in a fresh install.  So I can close the issue -- sorry!!
